TFUtil.py

```diff
@@ -1249,6 +1249,54 @@ def copy_template(self, name=None, dtype=None):
       kwargs["dtype"] = dtype
     return Data(**kwargs)
 
+  def copy_template_excluding_axis(self, exclude_axis, name=None):
+    """
+    :param str|None name: if set, this will be the new name.
+    :param int exclude_axis: axis to be removed.
+    :return: copy of myself excluding remove_axis_num axis, without placeholder.
+    :rtype: Data
+    """
+    kwargs = self.get_kwargs()
+    if exclude_axis < 0:
+      exclude_axis += self.batch_ndim
+      assert exclude_axis >= 0
+    assert 0 <= exclude_axis < self.batch_ndim
+    exclude_batch_dim_axis = exclude_axis == self.batch_dim_axis
+    if not exclude_batch_dim_axis:
+      axis_to_exclude_wo_b = self.get_batch_axis_excluding_batch(exclude_axis)
+    if exclude_axis == self.feature_dim_axis:
+      kwargs["dim"] = None
+
+    other_special_axes = self.get_special_axes_dict(
+      counted_with_batch_dim=True, only_available=True, include_batch_dim_axis=True)
+    for axis_name, axis in other_special_axes.items():
+      if axis == exclude_axis:
+        kwargs[axis_name] = None
+      else:
+        kwargs[axis_name] = axis if (axis < exclude_axis) else (axis - 1)
+
+    new_shape = list(self.shape)
+    if not exclude_batch_dim_axis:
+      del new_shape[axis_to_exclude_wo_b]
+
+    size_placeholder = None
+    if self.size_placeholder is not None:
+      if exclude_batch_dim_axis:
```

Comment:
Remove that `if`. You don't need to have conditional code here. Keep the code short if possible (easier to read).